modules/Bio/EnsEMBL/Hive/DBSQL/AccumulatorAdaptor.pm

```diff
@@ -62,6 +88,9 @@ sub fetch_structures_for_job_ids {
 
         ROW: while(my ($receiving_job_id, $struct_name, $key_signature, $stringified_value) = $sth->fetchrow_array() ) {
 
+            ($key_signature, $struct_name) = map {$self->check_and_dereference_analysis_data($_)}
+                ($key_signature, $struct_name);
+
```

Comment:
Do you need to "manually" call `check_and_dereference_analysis_data` here because it's the only place that does direct SQL access, whereas all the other adaptors naturally get the overflow working through fetch_all/slicer ?
Good catch !

Reply:
The other exception is in AnalysisJobAdaptor, which also does this check when it queries directly rather than delegating up to BaseAdaptor::fetch_all - which also should be updated to use check_and_dereference_analysis_data now that it's available.